lesson12/main.js

```diff
@@ -0,0 +1,85 @@
+const ul = document.getElementById("js-ul");
+const btn = document.getElementById("js-button")
+
+const renderLoading = () => {
+  const loading = document.getElementById("js-loading");
+  const loadingImg = document.createElement("img");
+  loadingImg.src = "img/loading.gif";
+  loadingImg.id = "loading-img";
+  loading.appendChild(loadingImg);
+};
+
+const removeLoading = () => {
+  document.getElementById("loading-img").remove();
+};
+
+const renderLists = (features) => {
+  const fragment = new DocumentFragment();
+
+  for (let feature of features) {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+    a.href = feature.to;
+    a.textContent = feature.text;
+    img.src = feature.img;
+    img.alt = feature.alt;
+
+    fragment
+      .appendChild(li)
+      .appendChild(a)
+      .insertAdjacentElement("afterbegin", img);
+  }
+
+  ul.appendChild(fragment);
+};
+
+const displayInfo = (info) => {
+  const li = document.createElement('li');
+  if (typeof info === 'string') {
+    li.textContent = info;
+    ul.appendChild(li);
+    return console.log(info);
+  }
+  throw new Error('Please pass a string value');
+};
+
+const displayErrorInfo = (error) => {
+  const li = document.createElement('li');
+  li.textContent = `error: ${JSON.stringify(error)}`;
+  ul.appendChild(li);
+};
+
+const getData = async (api) => {
+  let res = null;
+  try {
+    res = await fetch(api);
+  } catch (e) {
+    displayErrorInfo(e);
+  }
+  if (res.ok) {
+    return res.json();
+  } else {
+    displayErrorInfo(res);
+  }
+};
+
+const displayList = async () => {
+  let res = await getData("https://mocki.io/v1/ee8a871e-2b46-4a91-b565-4d6f9216f300");
+  if (res.data) {
+    renderLists(res.data);
+  } else {
+    displayInfo("no data");
+  }
+};
+
+
+const init = () => {
+  btn.remove();
+  renderLoading();
+  displayList();
+  removeLoading();
+};
```

Comment:
IMO
I think the load is released before the data acquisition is finished.
I have written code that uses setimeout to acquire data over a pseudo 5 second period. 
[DEMO](https://codesandbox.io/s/kind-shtern-qz7mv3?file=/main.js)

I think it is more appropriate to execute removeLoading after the data acquisition is finished.

Reply:
Thank you Sae-san for your feedback and even created a DEMO file. 
As of lesson 10, setTimeout is not required so that it is no visible that loading image is rendered and removed very well and did not notice that the loading image was removed before the fetched data returned in my original code. 

I fixed it to displayList() to call removeLoading() after fetching data in order for the await fetch() will be applied before executing removeLoaing(). 

Kindly check my latest commit